src/backend/joanie/tests/core/test_api_contract.py

```diff
@@ -621,3 +629,447 @@ def test_api_contract_download_authenticated_should_fail_if_contract_is_not_sign
+        the ZIP archive exists in storages.
+        """
+        user = factories.UserFactory()
+        organization = factories.OrganizationFactory()
+        factories.UserOrganizationAccessFactory(organization=organization, user=user)
+        token = self.get_user_token(user.username)
+        # Prepare ZIP archive in storage
+        zip_uuid = uuid4()
+        zip_archive_name = contract_utility.generate_zip_archive(
+            pdf_bytes_list=[b"content_1", b"content_2"],
+            user_uuid=user.id,
+            zip_uuid=zip_uuid,
+        )
+
+        response = self.client.get(
+            reverse("contracts-zip_archive", kwargs={"zip_id": str(zip_uuid)}),
+            HTTP_AUTHORIZATION=f"Bearer {token}",
+        )
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.headers["Content-Type"], "application/zip")
+        self.assertEqual(
+            response.headers["Content-Disposition"],
+            f'attachment; filename="{zip_archive_name}"',
+        )
+        # Clear the storage
+        storage = storages["contracts"]
+        storage.delete(zip_archive_name)
+
+    def test_api_contract_get_zip_archive_authenticated_simulate_waiting_for_zip_archive_ready(
+        self,
+    ):
+        """
+        Authenticated user should be able to GET his ZIP archive once it is ready.
+        We simulate in this test that the user requested his ZIP archive but it not yet
+        ready. First the response will return a 400 status code, and once it is ready, it
+        will return a status code 200 with the ZIP archive in the response.
+        """
+        user = factories.UserFactory()
+        organization = factories.OrganizationFactory()
+        factories.UserOrganizationAccessFactory(organization=organization, user=user)
+        zip_uuid = uuid4()
+        token = self.get_user_token(user.username)
+
+        response = self.client.get(
+            reverse("contracts-zip_archive", kwargs={"zip_id": str(zip_uuid)}),
+            HTTP_AUTHORIZATION=f"Bearer {token}",
+        )
+
+        self.assertEqual(response.status_code, 404)
+
+        # Prepare ZIP archive in storage
+        zip_archive_name = contract_utility.generate_zip_archive(
+            pdf_bytes_list=[b"content_1", b"content_2"],
+            user_uuid=user.id,
+            zip_uuid=zip_uuid,
+        )
+        response = self.client.get(
+            reverse("contracts-zip_archive", kwargs={"zip_id": str(zip_uuid)}),
+            HTTP_AUTHORIZATION=f"Bearer {token}",
+        )
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.headers["Content-Type"], "application/zip")
+        self.assertEqual(
+            response.headers["Content-Disposition"],
+            f'attachment; filename="{zip_archive_name}"',
+        )
+        # Clear the storage
+        storage = storages["contracts"]
+        storage.delete(zip_archive_name)
```

Comment:
Could done in the tear down method nope ?

Reply:
this is useless, we use the `InMemoryStorage`. I have to remove it.